tests/benchmark/conftest.py

```diff
@@ -7,6 +7,32 @@
 
 def pytest_collection_modifyitems(config, items):
     """Add the `benchmark` marker to all tests under `./tests/benchmark`."""
-    for item in items:
-        if Path(__file__).parent in Path(item.fspath).parents:
-            item.add_marker(pytest.mark.benchmark)
+    marker_expr = config.getoption("-m", default="")
+    gas_benchmark_values = config.getoption("--gas-benchmark-values", default=None)
+    run_benchmarks = marker_expr and (
+        "benchmark" in marker_expr and "not benchmark" not in marker_expr
+    )
+
+    if gas_benchmark_values:
+        run_benchmarks = True
+
+    items_for_removal = []
+    for i, item in enumerate(items):
+        is_benchmark_test = Path(__file__).parent in Path(item.fspath).parents
+
+        if is_benchmark_test:
+            benchmark_marker = pytest.mark.benchmark
+
+            if gas_benchmark_values:
+                gas_values = [int(v.strip()) for v in gas_benchmark_values.split(",")]
+                benchmark_marker = pytest.mark.benchmark(gas_values=gas_values)
```

Comment:
```suggestion
```
I'd lean towards not doing this here because the gas-benchmark-values are a properties of the current test filling execution, rather than the particular test itself.

Plus if we change this parameter's input format, we have to come to this file too to match it and we might miss it.